byzcoin/contract_deferred.go

```diff
@@ -400,6 +400,20 @@ func (c *contractDeferred) VerifyInstruction(rst ReadOnlyStateTrie, inst Instruc
 	return nil
 }
 
+// This function is used in the case we do a deferred transaction on a deferred
+// transaction.
+func (c *contractDeferred) VerifyDeferredInstruction(rst ReadOnlyStateTrie, inst Instruction, ctxHash []byte) error {
+	// We make a special case for the delete instruction. Anyone should be able
+	// to delete a deferred contract that has expired.
+	if inst.GetType() == DeleteType && uint64(rst.GetIndex()) >= c.DeferredData.ExpireBlockIndex {
+		return nil
+	}
+	if err := inst.VerifyWithOption(rst, ctxHash, false); err != nil {
+		return err
+	}
+	return nil
```

Comment:
this part can be simplified to `return inst.VerifyWithOption(rst, ctxHash, false)`.

Reply:
that's right, I forgot to wrap the errors.